### DISCUSSION TITLE
Does debounce work?

### DISCUSSION THREAD

**Discussion by quolpr** [nvimtools/none-ls.nvim#151] [Q&A]
I was searching through the code of none-ls, and didn't notice where `debounce` config is used.  I only see that it set to `debounce_text_changes`, but `debounce_text_changes` is not used at code too. 

https://github.com/search?q=repo%3Anvimtools%2Fnone-ls.nvim%20debounce&type=code

Does none-ls really respect debounce?

I have an issue with it - chatgpt live-code generator (like https://github.com/Robitx/gp.nvim ) changes buffer frequently, which spawning a lot of checks from none-ls.

**Comment by quolpr** [comment]
I see the commit from the past
https://github.com/nvimtools/none-ls.nvim/commit/310d49c909609bf58bfa44d36289dd1bf88f936e
But I can't find any connected code in current main with this commit

**Comment by mochaaP** [comment]
It is done by debouncing the `didChange` notification, which gp.nvim fires on every update (by calling `nvim_buf_set_lines`).

You can check `:h lsp` and neovim's source for more information.

A potential workaround is to add an autocmd to suppress LSP updates (e.g. detach the buffer) during the completion. Since null-ls is mostly stateless, this won't be a huge performance drawback.

**Reply by quolpr** [reply]
Got it, thanks. Could you show the particular code lines where debounce happen in none-ls? I mean I don't see where debounce variable is used in the nonde-ls code

**Reply by quolpr** [reply]
I still think that debounce actually doesn't setup correctly in none-ls. When I set it to even 10s, none-ls still triggers on each buffer change

**Reply by mochaaP** [reply]
This happens in neovim itself.
